## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 437178`
Commit: `6be0efbc906e6ff97c756ee3eccf974f835a262e`

---
### `x86_64-linux`
<details>
  <summary>:x: 12 packages failed to build:</summary>
  <ul>
    <li>langgraph-cli (python313Packages.langgraph-cli)</li>
    <li>langgraph-cli.dist (python313Packages.langgraph-cli.dist)</li>
    <li>python312Packages.langgraph</li>
    <li>python312Packages.langgraph-cli</li>
    <li>python312Packages.langgraph-cli.dist</li>
    <li>python312Packages.langgraph-runtime-inmem</li>
    <li>python312Packages.langgraph-runtime-inmem.dist</li>
    <li>python312Packages.langgraph.dist</li>
    <li>python313Packages.langgraph</li>
    <li>python313Packages.langgraph-runtime-inmem</li>
    <li>python313Packages.langgraph-runtime-inmem.dist</li>
    <li>python313Packages.langgraph.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python312Packages.langgraph-checkpoint-postgres</li>
    <li>python312Packages.langgraph-checkpoint-postgres.dist</li>
    <li>python312Packages.langgraph-prebuilt</li>
    <li>python312Packages.langgraph-prebuilt.dist</li>
    <li>python312Packages.langgraph-sdk</li>
    <li>python312Packages.langgraph-sdk.dist</li>
    <li>python313Packages.langgraph-checkpoint-postgres</li>
    <li>python313Packages.langgraph-checkpoint-postgres.dist</li>
    <li>python313Packages.langgraph-prebuilt</li>
    <li>python313Packages.langgraph-prebuilt.dist</li>
    <li>python313Packages.langgraph-sdk</li>
    <li>python313Packages.langgraph-sdk.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python312Packages.langgraph</summary>
<pre>CREATE ROLE
CREATE DATABASE
starting redis
waiting for redis to be ready
Could not connect to Valkey at /build/run/redis.sock: No such file or directory
pytest flags: -m pytest --ignore-glob=tests/test_checkpoint_migration.py --ignore-glob=tests/test_large_cases.py --ignore-glob=tests/test_large_cases_async.py --ignore-glob=tests/test_pregel.py --ignore-glob=tests/test_pregel_async.py -k not\ \(test_remote_graph_basic_invoke\)\ and\ not\ \(test_remote_graph_stream_messages_tuple\)\ and\ not\ \(test_cancel_graph_astream\)\ and\ not\ \(test_cancel_graph_astream_events_v2\)\ and\ not\ \(test_channel_values\)\ and\ not\ \(test_fork_always_re_runs_nodes\)\ and\ not\ \(test_interruption_without_state_updates\)\ and\ not\ \(test_interruption_without_state_updates_async\)\ and\ not\ \(test_invoke_two_processes_in_out_interrupt\)\ and\ not\ \(test_nested_graph_interrupts\)\ and\ not\ \(test_no_modifier_async\)\ and\ not\ \(test_no_modifier\)\ and\ not\ \(test_pending_writes_resume\)\ and\ not\ \(test_remove_message_via_state_update\) --numprocesses=24
ImportError while loading conftest '/build/source/libs/langgraph/tests/conftest.py'.
tests/conftest.py:11: in <module>
    from langgraph.cache.redis import RedisCache
E   ModuleNotFoundError: No module named 'langgraph.cache.redis'
stopping postgresql
waiting for server to shut down....2025-08-26 17:40:41.012 UTC [400] LOG:  received fast shutdown request
2025-08-26 17:40:41.024 UTC [400] LOG:  aborting any active transactions
2025-08-26 17:40:41.026 UTC [400] LOG:  background worker "logical replication launcher" (PID 406) exited with exit code 1
2025-08-26 17:40:41.026 UTC [401] LOG:  shutting down
2025-08-26 17:40:41.036 UTC [401] LOG:  checkpoint starting: shutdown immediate
2025-08-26 17:40:41.452 UTC [401] LOG:  checkpoint complete: wrote 934 buffers (5.7%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.029 s, sync=0.362 s, total=0.427 s; sync files=308, longest=0.009 s, average=0.002 s; distance=4259 kB, estimate=4259 kB; lsn=0/1915748, redo lsn=0/1915748
2025-08-26 17:40:41.459 UTC [400] LOG:  database system is shut down
 done
server stopped</pre>
</details>
<details>
<summary>python313Packages.langgraph</summary>
<pre>CREATE ROLE
CREATE DATABASE
starting redis
waiting for redis to be ready
Could not connect to Valkey at /build/run/redis.sock: No such file or directory
pytest flags: -m pytest --ignore-glob=tests/test_checkpoint_migration.py --ignore-glob=tests/test_large_cases.py --ignore-glob=tests/test_large_cases_async.py --ignore-glob=tests/test_pregel.py --ignore-glob=tests/test_pregel_async.py -k not\ \(test_remote_graph_basic_invoke\)\ and\ not\ \(test_remote_graph_stream_messages_tuple\)\ and\ not\ \(test_cancel_graph_astream\)\ and\ not\ \(test_cancel_graph_astream_events_v2\)\ and\ not\ \(test_channel_values\)\ and\ not\ \(test_fork_always_re_runs_nodes\)\ and\ not\ \(test_interruption_without_state_updates\)\ and\ not\ \(test_interruption_without_state_updates_async\)\ and\ not\ \(test_invoke_two_processes_in_out_interrupt\)\ and\ not\ \(test_nested_graph_interrupts\)\ and\ not\ \(test_no_modifier_async\)\ and\ not\ \(test_no_modifier\)\ and\ not\ \(test_pending_writes_resume\)\ and\ not\ \(test_remove_message_via_state_update\) --numprocesses=24
ImportError while loading conftest '/build/source/libs/langgraph/tests/conftest.py'.
tests/conftest.py:11: in <module>
    from langgraph.cache.redis import RedisCache
E   ModuleNotFoundError: No module named 'langgraph.cache.redis'
stopping postgresql
waiting for server to shut down....2025-08-26 17:40:40.800 UTC [400] LOG:  received fast shutdown request
2025-08-26 17:40:40.808 UTC [400] LOG:  aborting any active transactions
2025-08-26 17:40:40.810 UTC [400] LOG:  background worker "logical replication launcher" (PID 406) exited with exit code 1
2025-08-26 17:40:40.810 UTC [401] LOG:  shutting down
2025-08-26 17:40:40.816 UTC [401] LOG:  checkpoint starting: shutdown immediate
2025-08-26 17:40:41.282 UTC [401] LOG:  checkpoint complete: wrote 934 buffers (5.7%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.023 s, sync=0.420 s, total=0.473 s; sync files=308, longest=0.009 s, average=0.002 s; distance=4259 kB, estimate=4259 kB; lsn=0/1915748, redo lsn=0/1915748
2025-08-26 17:40:41.289 UTC [400] LOG:  database system is shut down
 done
server stopped</pre>
</details>
</details>

--- a/pkgs/development/python-modules/langgraph-checkpoint-sqlite/default.nix
+++ b/pkgs/development/python-modules/langgraph-checkpoint-sqlite/default.nix
@@ -74,8 +74,12 @@ buildPythonPackage rec {
     "test_search"
   ];
 
-  passthru.updateScript = gitUpdater {
-    rev-prefix = "checkpointsqlite==";
+  passthru = {
+    # python updater script sets the wrong tag
+    skipBulkUpdate = true;
+    updateScript = gitUpdater {
+      rev-prefix = "checkpointsqlite==";
+    };
   };
 
   meta = {

--- a/pkgs/development/python-modules/langgraph-cli/default.nix
+++ b/pkgs/development/python-modules/langgraph-cli/default.nix
@@ -24,14 +24,14 @@
 
 buildPythonPackage rec {
   pname = "langgraph-cli";
-  version = "2.1.1";
+  version = "0.3.6";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "langchain-ai";
     repo = "langgraph";
-    tag = "checkpoint==${version}";
-    hash = "sha256-UY3AChShKfOrtOQzOm5vi3Yy3rlBc+TAje9L2L6My/U=";
+    tag = "cli==${version}";
+    hash = "sha256-tBMdFOHSRjw0PtE19XytLU4MmjR3NBLJxUqWoG4L2F8=";
   };
 
   sourceRoot = "${src.name}/libs/cli";
@@ -80,8 +80,12 @@ buildPythonPackage rec {
     "test_build_command_shows_wolfi_warning"
   ];
 
-  passthru.updateScript = gitUpdater {
-    rev-prefix = "cli==";
+  passthru = {
+    # python updater script sets the wrong tag
+    skipBulkUpdate = true;
+    updateScript = gitUpdater {
+      rev-prefix = "cli==";
+    };
   };
 
   meta = {

--- a/pkgs/development/python-modules/langgraph/default.nix
+++ b/pkgs/development/python-modules/langgraph/default.nix
@@ -40,14 +40,14 @@
 }:
 buildPythonPackage rec {
   pname = "langgraph";
-  version = "2.1.1";
+  version = "0.6.4";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "langchain-ai";
     repo = "langgraph";
-    tag = "checkpoint==${version}";
-    hash = "sha256-UY3AChShKfOrtOQzOm5vi3Yy3rlBc+TAje9L2L6My/U=";
+    tag = version;
+    hash = "sha256-9jl16cKp3E7j79PXrr/3splrcJtfQQN7yFJ5sfa6c+I=";
   };
 
   postgresqlTestSetupPost = ''
@@ -133,11 +133,15 @@ buildPythonPackage rec {
   ];
 
   # Since `langgraph` is the only unprefixed package, we have to use an explicit match
-  passthru.updateScript = nix-update-script {
-    extraArgs = [
-      "--version-regex"
-      "([0-9.]+)"
-    ];
+  passthru = {
+    # python updater script sets the wrong tag
+    skipBulkUpdate = true;
+    updateScript = nix-update-script {
+      extraArgs = [
+        "--version-regex"
+        "([0-9.]+)"
+      ];
+    };
   };
 
   meta = {

